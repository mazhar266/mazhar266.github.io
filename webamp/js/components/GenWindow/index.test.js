import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import mockGetStore from "../../__mocks__/storeMock";

import { GenWindow } from "./index";

describe("GenWindow", () => {
  let store;
  beforeEach(() => {
    store = mockGetStore();
  });
  it("renders to snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <GenWindow
            title="My Window"
            selected
            close={() => {}}
            windowId="TEST_WINDOW_ID"
            scrollVolume={() => {}}
            windowSize={[0, 0]}
          >
            {() => {}}
          </GenWindow>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
