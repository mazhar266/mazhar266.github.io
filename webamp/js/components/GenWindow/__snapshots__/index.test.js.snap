// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenWindow renders to snapshot 1`] = `
<div
  onFocus={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
  tabIndex={-1}
>
  <div
    className="gen-window window selected"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <div
      className="gen-top draggable"
    >
      <div
        className="gen-top-left draggable"
      />
      <div
        className="gen-top-left-fill draggable"
      />
      <div
        className="gen-top-left-end draggable"
      />
      <div
        className="gen-top-title draggable"
      >
        <div
          className="draggable gen-text-letter gen-text-m"
        />
        <div
          className="draggable gen-text-letter gen-text-y"
        />
        <div
          className="draggable gen-text-letter gen-text-space"
        />
        <div
          className="draggable gen-text-letter gen-text-w"
        />
        <div
          className="draggable gen-text-letter gen-text-i"
        />
        <div
          className="draggable gen-text-letter gen-text-n"
        />
        <div
          className="draggable gen-text-letter gen-text-d"
        />
        <div
          className="draggable gen-text-letter gen-text-o"
        />
        <div
          className="draggable gen-text-letter gen-text-w"
        />
      </div>
      <div
        className="gen-top-right-end draggable"
      />
      <div
        className="gen-top-right-fill draggable"
      />
      <div
        className="gen-top-right draggable"
      >
        <div
          className="gen-close selected"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="gen-middle"
    >
      <div
        className="gen-middle-left draggable"
      >
        <div
          className="gen-middle-left-bottom draggable"
        />
      </div>
      <div
        className="gen-middle-center"
      />
      <div
        className="gen-middle-right draggable"
      >
        <div
          className="gen-middle-right-bottom draggable"
        />
      </div>
    </div>
    <div
      className="gen-bottom draggable"
    >
      <div
        className="gen-bottom-left draggable"
      />
      <div
        className="gen-bottom-right draggable"
      >
        <div
          id="gen-resize-target"
          onMouseDown={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
